File: tools/tal_support/as274_fc/F/t6.f
      program test6
c     Use the STEAM data from Draper & Smith to test PCORR.

      integer ier, case, col
      double precision d(10), rbar(45), thetab(10), sserr 
      double precision x(10), y, one, work(10), cormat(36), ycorr(10)
      data one/1.0/

      open(9, file='steam.data', status='old')
      call clear(10, 45, d, rbar, thetab, sserr, ier)

      x(1) = one
      do case = 1, 25
         read(9, *) (x(col),col=2,10), y
         call includ(10, 45, one, x, y, d, rbar, thetab, sserr, ier)
      end do
      write(*, *) 'ier = ', ier, ' and sserr = ', sserr

      call pcorr(10, 45, d, rbar, thetab, sserr, 1, work, cormat, 36,
     +      ycorr, ier)
      if (ier .ne. 0) write(*, *) 'pcorr() returned ', ier

      write(*, 900) (ycorr(col),col=1,10)
  900 format(' Correlations with the dependent variable:'/ 1x, 10f7.4/)
      write(*, 910) cormat
  910 format(' Correlations amongst the predictors:'/
     +       1x, '1.0', 3x, 8f8.4/
     +       9x, '1.0', 3x, 7f8.4/
     +      17x, '1.0', 3x, 6f8.4/
     +      25x, '1.0', 3x, 5f8.4/
     +      33x, '1.0', 3x, 4f8.4/
     +      41x, '1.0', 3x, 3f8.4/
     +      49x, '1.0', 3x, 2f8.4/
     +      57x, '1.0', 3x, f8.4/
     +      65x, '1.0')
      end

File: tools/tal_support/as274_fc/C/t6.c
#include <stdio.h>
#include "as274c.h"
#include "misc.h"

int main()
{
    double *d, *rbar, *thetab, sserr, **x, *y, *cormat, *ycorr;
    int error, i;

    x = dmatrix(0, 24, 0, 9); y = dvector(0, 24);
    d = dvector(0, 9);
    rbar = dvector(0, 44);
    thetab = dvector(0, 9);
    cormat = dvector(0, 35);
    ycorr = dvector(0, 9);

    error = clear(10, 45, d, rbar, thetab, &sserr);
    if (error) printf("clear() returned %d\n", error);
/* Grab data */
#include "steam.h"
    for (i=0; i<25; i++) {
        error = includ(10, 45, 1, x[i], y[i], d, rbar, thetab, &sserr);
        if (error) printf("includ() returned %d\n", error);
    }
    printf("SSerr = %.15g\n", sserr);
    error = pcorr(10, 45, d, rbar, thetab, &sserr, 1, cormat, 36, ycorr);
    if (error) printf("pcorr() returned %d\n", error);

    putdvec("ycorr", ycorr, 0, 9);
    pr_utdm_v(cormat, 8, 9, 4);
    return 0;
}
